mbrl/util/common.py

```diff
@@ -190,6 +194,116 @@ def create_replay_buffer(
+
+    Returns:
+        (tuple of :class:`mbrl.replay_buffer.TransitionIterator`): the training
+        and validation iterators, respectively.
+    """
+    data = replay_buffer.get_all(shuffle=True)
+    val_size = int(replay_buffer.num_stored * val_ratio)
+    train_size = replay_buffer.num_stored - val_size
+    train_data = data[:train_size]
+    train_iter = BootstrapIterator(
+        train_data,
+        batch_size,
+        ensemble_size,
+        shuffle_each_epoch=shuffle_each_epoch,
+        permute_indices=bootstrap_permutes,
+        rng=replay_buffer._rng,
+    )
+
+    val_iter = None
+    if val_size > 0:
+        val_data = data[train_size:]
+        val_iter = TransitionIterator(
+            val_data, batch_size, shuffle_each_epoch=False, rng=replay_buffer._rng
+        )
+
+    return train_iter, val_iter
+
+
+def get_sequence_buffer_iterator(
+    replay_buffer: ReplayBuffer,
+    batch_size: int,
+    val_ratio: float,
+    sequence_length: int,
+    ensemble_size: Optional[int] = None,
+    max_batches_per_loop: Optional[int] = None,
+) -> Tuple[SequenceTransitionIterator, Optional[SequenceTransitionIterator]]:
+    """Returns training/validation iterators for the data in the replay buffer.
+
+    Args:
+        replay_buffer (:class:`mbrl.util.ReplayBuffer`): the replay buffer from which
+            data will be sampled.
+        batch_size (int): the batch size for the iterators.
+        val_ratio (float): the proportion of data to use for validation. If 0., the
+            validation buffer will be set to ``None``.
+        sequence_length (int): the length of the sequences returned by the iterators.
+        ensemble_size (int): the number of models in the ensemble.
+        max_batches_per_loop (int, optional): if given, specifies how many batches
+            to return (at most) over a full loop of the iterator.
+
+    Returns:
+        (tuple of :class:`mbrl.replay_buffer.SequenceTransitionIterator`): the training
+        and validation iterators, respectively.
+    """
+
+    transitions = replay_buffer.get_all()
+    num_trajectories = len(replay_buffer.trajectory_indices)
+    val_size = int(num_trajectories * val_ratio)
+    train_size = num_trajectories - val_size
+    train_trajectories = replay_buffer.trajectory_indices[:train_size]
+
+    train_iterator = SequenceTransitionIterator(
+        transitions,
+        train_trajectories,
+        batch_size,
+        sequence_length,
+        ensemble_size,
+        rng=replay_buffer._rng,
+        max_batches_per_loop=max_batches_per_loop,
+    )
+
+    val_iterator = None
+    if val_size > 0:
+        val_trajectories = replay_buffer.trajectory_indices[train_size:]
+        val_iterator = SequenceTransitionIterator(
+            transitions,
+            val_trajectories,
+            batch_size,
+            sequence_length,
+            1,
+            rng=replay_buffer._rng,
+            max_batches_per_loop=max_batches_per_loop,
```

Comment:
```suggestion
            max_batches_per_loop=math.ceil(val_ratio * max_batches_per_loop),
```
In the current implementation ``val_ratio`` does not influence how many samples are used for evaluation if ``max_batches_per_loop`` is used (and the replay buffer contains enough data). Evaluation is then always done with ``max_batches_per_loop`` samples. With the changes above, one has more control over how many samples are actually used for evaluation. The value is rounded up to ensure that always at least one sample is used.

Reply:
I'll think a bit more about this (whether to incorporate this change or add a separate option), but I agree there should be more flexibility here.

Reply:
A separate option would also be fine for me

Reply:
Added a separate option in https://github.com/facebookresearch/mbrl-lib/pull/91/commits/892534118d773c91461ec1fcc059d9f60ce532b4.